expression/builtin.go

```diff
@@ -243,6 +243,26 @@ func newBaseBuiltinFuncWithTp(ctx sessionctx.Context, funcName string, args []Ex
 	return bf, nil
 }
 
+// newBaseBuiltinFuncWithFieldType create BaseBuiltinFunc with FieldType charset and collation.
+// do not check and compute collation.
+func newBaseBuiltinFuncWithFieldType(ctx sessionctx.Context, tp *types.FieldType, args []Expression) (baseBuiltinFunc, error) {
+	if ctx == nil {
+		panic("ctx should not be nil")
+	}
```

Comment:
Please return an error here for safety instead of causing a panic.

Reply:
other two function `newBaseBuiltinFunc` `newBaseBuiltinFuncWithTp` will panic if `ctx` is `nil`, should we change them too?

Reply:
Yes, I prefer to change them too, thanks~